File: apps/macos/NowFocusCore/EnforcementHealth.swift
import Foundation

public enum EnforcementStatus: String, Codable {
    case active
    case degraded
    case unavailable
}

public struct PlatformCapabilities: Codable {
    public let canBlockWebsites: Bool
    public let canBlockApplications: Bool
    public let isPrivilegedDaemonRunning: Bool
    
    public init(canBlockWebsites: Bool, canBlockApplications: Bool, isPrivilegedDaemonRunning: Bool) {
        self.canBlockWebsites = canBlockWebsites
        self.canBlockApplications = canBlockApplications
        self.isPrivilegedDaemonRunning = isPrivilegedDaemonRunning
    }
}

File: apps/windows/core/src/enforcement_health.rs
use serde::{Deserialize, Serialize};

/// Port of apps/macos/NowFocusCore/EnforcementHealth.swift. Note this is the
/// real 3-field `PlatformCapabilities` that shipped, not the 9-field version
/// sketched in focus_app_technical_architecture.md §10 — match what's
/// actually built, not the aspirational spec.
#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum EnforcementStatus {
    Unknown,
    Active,
    Degraded,
    Unavailable,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub struct PlatformCapabilities {
    pub can_block_websites: bool,
    pub can_block_applications: bool,
    pub is_privileged_service_running: bool,
}
